include/devicetree.h

```diff
@@ -701,6 +702,61 @@
 	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
 		    (DT_ENUM_IDX(node_id, prop)), (default_idx_value))
 
+/**
+ * @brief Get a string property's value as a token.
+ *
+ * This removes "the quotes" from string-valued properties, and converts
+ * non-alphanumeric characters to underscores. That can be useful, for example,
+ * when programmatically using the value to form a C variable or code.
+ *
+ * DT_STRING_TOKEN() can only be used for properties with string type.
+ *
+ * It is an error to use DT_STRING_TOKEN() in other circumstances.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             prop = "foo";
+ *     };
+ *     n2: node-2 {
+ *             prop = "FOO";
+ *     }
+ *     n3: node-3 {
+ *             prop = "123 foo";
+ *     };
+ *
+ * Example bindings fragment:
+ *
+ *     properties:
+ *       prop:
+ *         type: string
+ *
+ * Example usage:
+ *
+ *     DT_STRING_TOKEN((DT_NODELABEL(n1), prop) // foo
+ *     DT_STRING_TOKEN((DT_NODELABEL(n2), prop) // FOO
+ *     DT_STRING_TOKEN((DT_NODELABEL(n3), prop) // 123_foo
```

Comment:
Can you please remove the extra parentheses from this? I posted a fix for the existing macro here: https://github.com/zephyrproject-rtos/zephyr/pull/35640